src/weather.cpp

```diff
@@ -289,7 +289,8 @@ double trap::funnel_turns_per_charge( double rain_depth_mm_per_hour ) const
         return 0.0;
     }
 
-    const double surface_area_mm2 = M_PI * ( funnel_radius_mm * funnel_radius_mm );
+    const double surface_area_mm2 = static_cast<double>( M_PI * ( funnel_radius_mm *
```

Comment:
This doesn't make sense.  `M_PI` is surely already a `double`, so this whole expression should be already, without a cast?